src/main/java/com/amplitude/Response.java

```diff
@@ -10,9 +12,14 @@
   protected JSONObject invalidRequestBody;
   protected JSONObject rateLimitBody;
 
-  protected static Response populateResponse(JSONObject json) {
+  protected static Response populateResponse(JSONObject json) throws AmplitudeInvalidAPIKeyException {
     Response res = new Response();
     int code = json.getInt("code");
+    String invalidAPIKeyError = "Invalid API key: .*";
```

Comment:
I guess this was tested, and maybe the check should be under the certain status block?